### ISSUE TITLE
C# sources with `None` build action are not editable

### ISSUE THREAD

**Issue by atsushieno** [mono/monodevelop#4173]
Sometimes we add C# sources with build action `None ` (e.g. to become some kind of resources). The still need to be editable, but MonoDevelop fails to make it so. Cursors don't move when we switch to editor for such files.

Repro:
- Create a console project.
- Set build action for `Program.cs` from `Compile` to `None`.
- Close the editor.
- Open the file again.
- Try to edit it.

Checked with monodevelop 036b03b.

> VS bug [#583940](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/583940)

**Comment by mrward** [comment]
Reproduced on VS Mac 7.5 build 843 (d15-7).

VS Mac 7.4 build 1033 (d15-6) is OK.

The IDE log has:

```
ERROR [2018-03-15 09:22:44Z]: Error while updating the indentation engine
System.NullReferenceException: Object reference not set to an instance of an object
  at MonoDevelop.CSharp.Formatting.CSharpTextEditorIndentation.SafeUpdateIndentEngine (System.Int32 offset) [0x00000] in /Users/builder/data/lanes/5965/0f413f3d/source/monodevelop/main/src/addins/CSharpBinding/MonoDevelop.CSharp.Formatting/CSharpTextEditorIndentation.cs:84 
ERROR [2018-03-15 09:22:44Z]: Error in text editor extension chain
System.NullReferenceException: Object reference not set to an instance of an object
  at MonoDevelop.CSharp.Formatting.CSharpTextEditorIndentation.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x002c0] in /Users/builder/data/lanes/5965/0f413f3d/source/monodevelop/main/src/addins/CSharpBinding/MonoDevelop.CSharp.Formatting/CSharpTextEditorIndentation.cs:436 
  at MonoDevelop.Ide.Editor.Extension.TextEditorExtension.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x00000] in /Users/builder/data/lanes/5965/0f413f3d/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/TextEditorExtension.cs:83 
  at MonoDevelop.DocFood.DocFoodTextEditorExtension.KeyPress (MonoDevelop.Ide.Editor.Extension.KeyDescriptor descriptor) [0x0000a] in /Users/builder/data/lanes/5965/0f413f3d/source/monodevelop/main/src/addins/MonoDevelop.DocFood/MonoDevelop.DocFood/DocFoodTextEditorExtension.cs:59 
  at MonoDevelop.SourceEditor.ExtensibleTextEditor.ExtensionKeyPress (Gdk.Key key, System.UInt32 ch, Gdk.ModifierType state) [0x0003a] in /Users/builder/data/lanes/5965/0f413f3d/source/monodevelop/main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor/ExtensibleTextEditor.cs:282 
```

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.5.0.917 (d15-7)

Pull Request #4199 merged by: Mike Krüger
Author: mono
Commit: 1bd1595ce75a2443bd03fee64b1fd11d536020c8 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.6.0.159 (master)

Author: Mike Krüger
Commit: 8115924dc412ba1e985b93e1d6d2e583f57196b7 (mono/monodevelop)
